src/farcasterd/trade_state_machine.rs

```diff
@@ -212,70 +214,91 @@ impl From<ConsumedDealRole> for TradeRole {
 
 impl StateMachine<Runtime, Error> for TradeStateMachine {
     fn next(self, event: Event, runtime: &mut Runtime) -> Result<Option<Self>, Error> {
-        debug!(
-            "{} | Checking event request {} from {} for state transition",
-            self.swap_id().map_or("…".to_string(), |s| s.to_string()),
-            event.request,
-            event.source
-        );
+        let log_helper = self.log_helper();
+        log_helper.log_debug(format!(
+            "Checking event request {} from {} for state transition",
+            event.request, event.source
+        ));
         match self {
             TradeStateMachine::StartTaker => {
-                attempt_transition_to_taker_connect_or_take_deal(event, runtime)
+                attempt_transition_to_taker_connect_or_take_deal(event, runtime, log_helper)
```

Comment:
I don't see how to improve this while keeping the fns as is. But I'm fine going that direction...